packages/react-components/lib/reports/log-management/custom-lookup-filter.tsx

```diff
@@ -22,72 +20,33 @@ const MenuProps = {
   },
 };
 
-export interface CustomLookupFilterParserProps {
-  // Prop RowData from material-table is not exported so unknown
-  columnDef: Column<{ level: string; message: string; created: string; container: ContainerType }>;
-  onFilterChanged: (rowId: string, value: string | number | unknown) => void;
-}
-
-/**
- * Component created to override the default lookup filter.
- */
-
-export const CustomLookupFilterParser = (
-  props: CustomLookupFilterParserProps,
-): React.ReactElement => {
-  const { columnDef: columnDefRaw, onFilterChanged } = props;
-  /**
-   * The column type in the material-table library it's not working correctly. So as a workaround I'm assigning the types to the properties need it in this component.
-   */
-  const columnDef = columnDefRaw as {
-    tableData: EditCellColumnDef['tableData'];
-    filterOnItemSelect: unknown;
-    lookup: Record<string, string>;
-  };
-
-  return (
-    <CustomLookupFilter
-      tableId={columnDef.tableData.id}
-      filterValue={columnDef.tableData.filterValue}
-      filterOnItemSelect={columnDef.filterOnItemSelect}
-      onFilterChanged={onFilterChanged}
-      lookup={columnDef.lookup}
-    />
-  );
-};
-
 export interface CustomLookupFilterProps {
-  tableId: number;
-  filterValue: EditCellColumnDef['tableData']['filterValue'];
-  filterOnItemSelect: unknown;
-  onFilterChanged: (rowId: string, value: string | number | unknown) => void;
   lookup: Record<string, string>;
+  selectedFilter: string[];
+  setSelectedFilter: React.Dispatch<React.SetStateAction<string[]>>;
 }
 
 export const CustomLookupFilter = (props: CustomLookupFilterProps): React.ReactElement => {
-  const { tableId, filterValue, filterOnItemSelect, onFilterChanged, lookup } = props;
-  const [selectedFilter, setSelectedFilter] = React.useState(filterValue || []);
-  React.useEffect(() => {
-    setSelectedFilter(filterValue || []);
-  }, [filterValue]);
-
+  const { lookup, selectedFilter, setSelectedFilter } = props;
+  const [filterTracker, setFilterTracker] = React.useState<string[]>(selectedFilter);
```

Comment:
This state seems useless and introduces potential bugs. If `selectedFilter` changes, `filterTracker` will not be updated.